### ISSUE TITLE
Good training output, but Poor Testing Quality, and Input label become All Green?

### ISSUE THREAD

**Issue by aldhosutra** [NVlabs/SPADE#96]
Hello, thanks for this amazing work.

I want to play around with SPADE, and try to create my own dataset, it consist of training image and training label (no instance). The image looks like this:

Label: (PNG - Grayscale 8 Bit)
![01 - skandinavia modern (1)](https://user-images.githubusercontent.com/11401791/71465397-1e76f180-27ef-11ea-8212-81df839d90cd.png)

Image: (JPG - RGB 8 Bit)
![01 - skandinavia modern](https://user-images.githubusercontent.com/11401791/71465484-5bdb7f00-27ef-11ea-8d78-6bb4b192a57b.jpg)

I try to do training, by using this command:
`python train.py --name [mydataset] --dataset_mode custom --dataroot [mydataset] --label_dir [mydataset]/train_label --image_dir [mydataset]/train_img --no_instance --contain_dontcare_label --label_nc 24 --no_flip --preprocess_mode none --load_size 256 --crop_size 256 --display_freq 5000 --save_epoch_freq 100 --niter 2000`

In my checkpoint folder, i see the result are very good, here are some example:

Real Image:
![epoch968_iter90000_real_image](https://user-images.githubusercontent.com/11401791/71465653-cf7d8c00-27ef-11ea-982a-7f73c8bf2eb8.png)

Synthethized Image:
![epoch968_iter90000_synthesized_image](https://user-images.githubusercontent.com/11401791/71465671-d73d3080-27ef-11ea-8f52-1ce90ab7d0bf.png)

But, despite of excellent training checkpoint output, my test result are very bad, i use exactly same label to do testing, and i use this command to do testing:
`python test.py --name [mydataset] --dataset_mode custom --dataroot [mydataset] --label_dir [mydataset]/train_label --image_dir [mydataset]/train_img --no_instance --contain_dontcare_label --label_nc 24 --load_size 256 --crop_size 256`

And here is the example of testing output:

![01 - skandinavia modern](https://user-images.githubusercontent.com/11401791/71465796-1ec3bc80-27f0-11ea-9e5e-3a076a729c2c.png)

Why is this happening?

I also find out that, the input image displayed in training epoch output, and in ./results/[mydataset]/test_latest/images/input_label/ in testing phase is all green, like this:
![epoch1001_iter93044_input_label](https://user-images.githubusercontent.com/11401791/71465885-621e2b00-27f0-11ea-9163-416d537a16c5.png)

is there something wrong about my dataset format?

Thanks for your kind response :)

**Comment by aldhosutra** [comment]
It looks like i use wrong color table for labeling, when i try to follow cityscapes label color, and apply to my datasets, it works. I feel perfectly dumb right now :'(

Will continue to explore and experiment SPADE. Thanks

Solved, closed

**Comment by koiloi** [comment]
> It looks like i use wrong color table for labeling

@aldhosutra  Could you please tell me where exactly did you changed the color table for labeling? I'm having the same issue like you had but I don't know the solution yet. Probably your modification may help me solve it but I don't know which part of the code you're talking about. Thanks.

**Comment by jmssxgp** [comment]
i have the same question, you can comment the model.eval() in test.py and try again.
